java-checks/src/main/java/org/sonar/java/checks/MutableMembersUsageCheck.java

```diff
@@ -149,18 +149,21 @@ private static boolean isThis(ExpressionTree expression) {
     return expression.is(Tree.Kind.IDENTIFIER) && ((IdentifierTree) expression).name().equals("this");
   }
 
-  private static boolean isImmutableFinalVariable(Symbol.VariableSymbol symbol) {
-    if (symbol.isFinal()) {
-      VariableTree declaration = symbol.declaration();
-      // symbol is private, so declaration can only be null if assignment is done in static block
```

Comment:
> // symbol is private, so declaration can only be null if assignment is done in static block

I did not manage to reproduce a case where the declaration is null, but I still added a null check because I was not convinced by this comment. 
I think it makes sense to keep it for defensive programming.

Reply:
Maybe Objects.requireNonNull will be a better option here, cause I can't imagine a situation when declaration is null for private fields. But null check is also fine for me

Reply:
Both alternatives sound good to me aswell. The behavior when the declaration is null (for whatever reason) is to proceed with the check of the usages as if it was not initialized, so it seems fine to me to keep it as it is.